data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java

```diff
@@ -32,53 +34,102 @@
 
     protected RemoteWebDriver driver;
     protected Logger logger;
-    private String cinemaName;
     private TmdbDataRetriever tmdb;
 
-    public AbstractCinemaScraper(String cinemaName) {
-        this.cinemaName = cinemaName;
+    protected Cinema cinema;
+
+    public AbstractCinemaScraper() {
         tmdb = new TmdbDataRetriever();
     }
 
     public void scrape() {
         logger = getLogger();
-        logger.info(String.format("Initializing Webdriver for scraper: [%s]", cinemaName));
+        cinema = saveObject(getCinema(), Cinema.class);
+
+        logger.info(String.format("Initializing Webdriver for scraper: [%s]", cinema.getName()));
         initlializeWebdriver();
 
-        logger.info(String.format("Start gathering data from: [%s]", cinemaName));
+        logger.info(String.format("Start gathering data from: [%s]", cinema.getName()));
         Stopwatch watch = Stopwatch.createStarted();
-        gatherData();
-        logger.info(String.format("Finished gathering data from: [%s] in %s milliseconds", cinemaName,
+        GatheringResult result = gatherData();
+        logger.info(String.format("Finished gathering data from: [%s] in %s milliseconds", cinema.getName(),
                 watch.elapsed(TimeUnit.MILLISECONDS)));
 
+        //get information for and save movies
+        //we also adjust the corresponding playlists here
+        processMovies(result);
+
+        //Reset playlist
+        logger.info(String.format("Resetting playlist for scraper: [%s]", cinema.getName()));
+        deletePlaylistFuture(cinema);
+
+        //adjust the cinema in playlist and save it
+        processPlaylists(result);
+
         driver.quit();
     }
 
+    private void processPlaylists(GatheringResult result) {
+        for (Playlist playlist : result.getPlaylists()) {
+            playlist.setCinema(cinema);
+            saveObject(playlist, Playlist.class);
+        }
+    }
+
+    private void processMovies(GatheringResult result) {
+        for (Movie movie : result.getMovies()) {
+            Movie filledMovie = retrieveMovieInformation(movie);
+            if (filledMovie != null) {
+                filledMovie = saveObject(filledMovie, Movie.class);
+                for (Playlist playlist : result.getPlaylists()) {
+                    //We want to compare for the specific instance here as its the only thing we have
+                    if (playlist.getMovie() == movie) {
+                        playlist.setMovie(filledMovie);
+                    }
+                }
+            } else {
+                Iterator<Playlist> iterator = result.getPlaylists().iterator();
+                while (iterator.hasNext()) {
+                    Playlist playlist = iterator.next();
+                    if (playlist.getMovie() == movie) {
+                        iterator.remove();
+                    }
+                }
+            }
+        }
+    }
```

Comment:
TBH I think the whole dance with replacing movies in the playlist is pretty ugly. But that's what we get with the pattern. 

Could probably be made a little bit prettier with some Java 8 streams magic but this is fine, too.